### PR TITLE
Adjust table group by styles and fix affected component

### PR DESCRIPTION
**Pull request by Christian862**
## Description
Adjusts the table display and size styles to remove the expandable detail horizontal scrollbar for each row, and allow the detail content to overflow, giving the main table the horizontal scrollbar. 

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: projects/components/src/table/cells/expanded-detail/table-expanded-detail-row-cell-renderer.component.scss [hunk 1]
```diff
@@ -2,7 +2,7 @@
 @import 'font';
 
 .table-expanded-cell-renderer {
-  padding: 0 0 16px 18px; // Using 18px for the right padding aligns the dotted line with the expander arrow
```
Comment (anandtiwary) [inline projects/components/src/table/cells/expanded-detail/table-expanded-detail-row-cell-renderer.component.scss:5]: does it still look nice?

File: projects/components/src/table/table.component.scss [hunk 2]
```diff
@@ -71,6 +76,14 @@ $header-height: 32px;
   }
 }
 
+.expandable-row {
```
Comment (anandtiwary) [inline projects/components/src/table/table.component.scss:79]: We need to test this on all pages a table is used. @Christian862 @itssharmasandeep could you guys please test this?

**Review by anandtiwary — COMMENTED** [review]
Change looks good, but we need to thoroughly test this before merge

File: projects/components/src/table/cells/expanded-detail/table-expanded-detail-row-cell-renderer.component.scss (hunk 1 above)
Comment (Christian862) [reply projects/components/src/table/cells/expanded-detail/table-expanded-detail-row-cell-renderer.component.scss:5]: imo yes, you can check the video sent in slack

File: projects/components/src/table/table.component.ts [hunk 3]
```diff
@@ -60,13 +60,13 @@ import { TableColumnConfigExtended, TableService } from './table.service';
...
   template: `
-    <div class="table">
+    <div class="table" [ngClass]="this.mode">
       <cdk-table
         *ngIf="this.dataSource"
         #cdkTable
         [multiTemplateDataRows]="this.isDetailType()"
         [dataSource]="this.dataSource"
-        [ngClass]="[this.display, this.pageable && this.isTableFullPage ? 'bottom-margin' : '']"
+        [ngClass]="[this.display, this.pageable && this.isTableFullPage ? 'bottom-margin' : '', this.mode]"
```
Comment (itssharmasandeep) [inline projects/components/src/table/table.component.ts:69]: I guess we do not need this here, Didn't see any usage of this

**Review by itssharmasandeep — APPROVED** [review]
LGTM. Tested it. Couldn't find any issues.

**Pull request merged**